contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/DrtAnalysisControlerListener.java

```diff
@@ -453,6 +445,35 @@ private static Map<Double, List<DrtLeg>> splitLegsIntoBins(Collection<DrtLeg> le
 		return splitLegs;
 	}
 
+	private static Map<Double, List<EventSequence>> splitEventsIntoBins(DrtEventSequenceCollector drtEventSequenceCollector, int startTime, int endTime, int binSize_s) {
+		// get all rejection events
+		Collection<EventSequence> rejectionEvents = drtEventSequenceCollector.getRejectedRequestSequences().values();
+
+		Map<Double, List<EventSequence>> rejections = new TreeMap<>();
+
+		for (int time = startTime; time < endTime; time = time + binSize_s) {
+
+			// rejection list in this timebin
+			List<EventSequence> rejectionList = new ArrayList<>();
+
+			//Iterate through each rejection
+			for (EventSequence seq : rejectionEvents){
+				//get rejection time
+				DrtRequestSubmittedEvent submission = seq.getSubmitted();
```

Comment:
But EventSequence seq is actually from rejection events. That means those submission event time are also only the rejection event time.

Reply:
The 'problem' is that for some technical reasons, the rejection could follow the submission event after some time passed. For example, a person submits a request at time t=0, the oprator tries to assign a vehicle, say for 10 seconds, and then rejects at time=10.

So this would roughly look like
```
<submissionEvent time="0" person="bla">
<rejectionEvent time="10" person="bla">
```

Then you would take t=0s for determing the time slot of the rejection, which is the time of the submission, not the rejection time.